src/storage.h

```diff
@@ -0,0 +1,273 @@
+  /// @param time the time of the tick  
+  virtual void Tick();
+
+  /// The handleTick function specific to the Storage.
+  /// @param time the time of the tock
+  virtual void Tock();
+
+  /// @brief The Storage request Materials of its given
+  /// commodity.
+  virtual std::set<cyclus::RequestPortfolio<cyclus::Material>::Ptr> GetMatlRequests();
+
+  /// @brief The Storage place accepted trade Materials in their
+  /// Inventory
+  virtual void AcceptMatlTrades(
+      const std::vector< std::pair<cyclus::Trade<cyclus::Material>,
+      cyclus::Material::Ptr> >& responses);
+
+  /// @brief Responds to each request for this facility's commodity.  If a given
+  /// request is more than this facility's inventory capacity, it will
+  /// offer its minimum of its capacities.
+  virtual std::set<cyclus::BidPortfolio<cyclus::Material>::Ptr>
+      GetMatlBids(cyclus::CommodMap<cyclus::Material>::type&
+                  commod_requests);
+
+  /// @brief respond to each trade with a material of out_commod and out_recipe
+  ///
+  /// @param trades all trades in which this trader is the supplier
+  /// @param responses a container to populate with responses to each trade
+  virtual void GetMatlTrades(
+    const std::vector< cyclus::Trade<cyclus::Material> >& trades,
+    std::vector<std::pair<cyclus::Trade<cyclus::Material>,
+    cyclus::Material::Ptr> >& responses);
+
+  /* --- */
+
+  /* --- Storage Members --- */
+
+  /* --- */
+  /// @brief the processing time required for a full process
+  inline void process_time_(int t) { process_time = t; }
+  inline int process_time_() const { return process_time; }
+
+  /// @brief the maximum amount allowed in inventory
+  inline void max_inv_size_(double c) { max_inv_size = c; }
+  inline double max_inv_size_() const { return max_inv_size; }
+
+  /// @brief the maximum amount processed per timestep
+  inline void capacity_(double c) { capacity = c; }
+  inline double capacity_() const { return capacity; }
+
+  /// @brief the cost per unit out_commod
+  inline void cost_(double c) { cost = c; }
+  inline double cost_() const { return cost; }
+
+  /// @brief the in commodity
+  inline void in_commod_(std::string c) { in_commod = c; }
+  inline std::string in_commod_() const { return in_commod; }
+
+  /// @brief the out commodity
+  inline void out_commod_(std::string c) { out_commod = c; }
+  inline std::string out_commod_() const { return out_commod; }
+
+  /// @brief the in recipe
+  inline void in_recipe_(std::string c) { in_recipe = c; }
+  inline std::string in_recipe_() const { return in_recipe; }
+
+  /// @brief the out recipe
+  inline void out_recipe_(std::string c) { out_recipe = c; }
+  inline std::string out_recipe_() const { return out_recipe; }
+
+  /// @brief current maximum amount that can be added to processing
+  inline double current_capacity() const { 
+    return std::min(capacity, max_inv_size - inventory.quantity()); }
+
+  /// @brief returns the time key for ready materials
```

Comment:
I remember this ready() function not being clear to you guyes when steve first saw this code. Maybe the documentation for this function would be clearer if we use the word index instead of key? Maybe not necessary. What do you think?